electrumsv/tests/test_wallet.py

```diff
@@ -550,64 +621,249 @@ def test_legacy_wallet_loading(mock_wallet_app_state, storage_info: WalletStorag
     future.result(5)
     # Wait for the done callback to finish.
     update_completion_event.wait()
 
     if "standard" == expected_type:
-        check_legacy_parent_of_standard_wallet(wallet, password=password,
+        check_legacy_parent_of_standard_wallet(wallet, password=new_password,
             add_indefinite_credential_mock=add_indefinite_credential_mock)
     elif "imported" == expected_type:
         if "privkey" in wallet_filename:
-            check_legacy_parent_of_imported_privkey_wallet(wallet, password, prv_keypairs)
+            check_legacy_parent_of_imported_privkey_wallet(wallet, new_password,
+                keyinstance_private_keys)
         elif "address" in expected_subtypes:
             check_legacy_parent_of_imported_address_wallet(wallet)
         else:
             raise Exception(f"unrecognised wallet file {wallet_filename}")
     elif "multisig" == expected_type:
-        check_legacy_parent_of_multisig_wallet(wallet, password)
+        check_legacy_parent_of_multisig_wallet(wallet, new_password)
     elif "hardware" == expected_type:
         check_legacy_parent_of_hardware_wallet(wallet)
     elif "blank" == expected_type:
         check_parent_of_blank_wallet(wallet)
     else:
         raise Exception(f"unrecognised wallet file {wallet_filename}")
 
-    check_specific_wallets(wallet, password, storage_info)
+    check_specific_wallets(wallet, new_password, storage_info)
 
-    if expected_network in { "testnet", "regtest" }:
-        Net.set_to(SVMainnet)
+    ## Cached data AFTER the password update
+    password_token_after_cached = wallet.get_storage().get('password-token')
+    assert pw_decode(password_token_after_cached, new_password) is not None
+    assert pw_decode(password_token_before_cached, initial_password) is not None
+    test_state.count_of_password_token_reencryptions_cache += 1
 
+    if "imported" == expected_type and "privkey" in wallet_filename:
+        keyinstance_private_keys_cached_after = get_cached_imported_privkeys(new_password)
+        assert keyinstance_private_keys_cached_after == keyinstance_private_keys_cached
+        test_state.count_of_keyinstance_privkey_reencryptions_cache += \
+            len(keyinstance_private_keys_cached_after)
+
+    encrypted_api_keys_cached_after = get_cached_server_api_keys(new_password)
+    assert encrypted_api_keys_cached_after == encrypted_api_keys_cached
+    test_state.count_of_server_api_key_reencryptions_cache += \
+        len(encrypted_api_keys_cached)
+
+    master_key_xprvs_cached_after, master_key_seeds_cached_after, \
+        master_key_passphrases_cached_after = get_cached_master_keys_encrypted_data(new_password)
+    assert master_key_xprvs_cached_after == master_key_xprvs_cached
+    assert master_key_seeds_cached_after == master_key_seeds_cached
+    assert master_key_passphrases_cached_after == master_key_passphrases_cached
+    test_state.count_of_masterkeys_table_xprv_reencryptions_cache += \
```

Comment:
Please listen to me. The point of these counts is to know how many wallets had a case we want to know we checked for, and to be able to validate it was at least one. What you are counting here is a strange deviation from that requirement. Please correct it. Count the wallets that were relevant to a type of check we are making like the original requirements.

Reply:
Ok. So `if len(...) > 0: increment count by 1`